src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/CourtIntegrationTest.kt

```diff
@@ -209,189 +203,218 @@ class CourtIntegrationTest : IntegrationTest() {
     @Test
     fun `should not find booking when only pre and post appointments exist`() {
 
-      whenever(videoLinkBookingRepository.findById(videoBookingId)).thenReturn(Optional.of(theVideoLinkBooking))
+      val videoLinkBookingId = videoLinkBookingRepository.save(makeVideoLinkBooking()).id!!
 
-      prisonApiMockServer.stubGetPrisonAppointment(preAppointmentId, objectMapper.writeValueAsString(preAppointment))
-      prisonApiMockServer.stubGetPrisonAppointment(postAppointmentId, objectMapper.writeValueAsString(postAppointment))
+      prisonApiMockServer.stubGetPrisonAppointment(
+        preAppointmentId,
+        objectMapper.writeValueAsString(prePrisonAppointment)
+      )
+      prisonApiMockServer.stubGetPrisonAppointment(
+        postAppointmentId,
+        objectMapper.writeValueAsString(postPrisonAppointment)
+      )
 
       webTestClient.get()
-        .uri("/court/video-link-bookings/$videoBookingId")
+        .uri("/court/video-link-bookings/$videoLinkBookingId")
         .headers(setHeaders())
         .exchange()
         .expectStatus().isNotFound
     }
   }
 
-  @Test
-  fun `should return video link appointment by appointment id`() {
-    whenever(videoLinkAppointmentRepository.findVideoLinkAppointmentByAppointmentIdIn(setOf(1L)))
-      .thenReturn(
-        setOf(
-          VideoLinkAppointment(
-            id = 1L,
-            bookingId = 1L,
-            appointmentId = 1L,
-            hearingType = HearingType.PRE,
-            court = "York",
-            courtId = "TSTCRT"
-          )
-        )
+  @Nested
+  inner class GetAppointments {
+    @BeforeEach
+    fun resetVideoLinkBookings() {
+      JdbcTestUtils.deleteFromTables(jdbcTemplate, "VIDEO_LINK_APPOINTMENT", "VIDEO_LINK_BOOKING")
+    }
+
+    @Test
+    fun `should return video link appointment by appointment id`() {
+      val persistentBooking = videoLinkBookingRepository.save(
+        VideoLinkBooking(
+          offenderBookingId = 1L,
+          courtName = "York",
+          courtId = "TSTCRT"
+        ).apply { addMainAppointment(appointmentId = 1L) }
       )
 
-    webTestClient.post()
-      .uri("/court/video-link-appointments")
-      .headers(setHeaders())
-      .bodyValue(setOf(1L))
-      .exchange()
-      .expectStatus().isOk
-      .expectBody()
-      .json(
-        """
+      val mainAppointmentId = persistentBooking.appointments[MAIN]?.id
+      assertThat(mainAppointmentId).isNotNull
+
+      webTestClient.post()
+        .uri("/court/video-link-appointments")
+        .headers(setHeaders())
+        .bodyValue(setOf(1L))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .json(
+          """
```

Comment:
Slight bit of inconsistency here as we normally compare against a .json resource file.